Source/radiation/rad_util.H

```diff
@@ -1,12 +1,56 @@
-#include <cmath>
-
 #ifndef CASTRO_RAD_UTIL_H
 #define CASTRO_RAD_UTIL_H
 
+#include <cmath>
+
+#include <Castro_util.H>
+
 #include <fluxlimiter.H>
 
 using namespace amrex;
 
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void cell_center_metric(int i, int j, int k, const GeometryData& geomdata,
+                        Real& r, Real& s)
+{
+    if (geomdata.Coord() == 0)
+    {
+        r = 1.0_rt;
+        s = 1.0_rt;
+    }
+    else if (geomdata.Coord() == 1)
+    {
+        GpuArray<Real, 3> loc;
+        position(i, j, k, geomdata, loc);
+
+        r = loc[0];
+        s = 1.0_rt;
+    }
+    else if (geomdata.Coord() == 2)
+    {
+       GpuArray<Real, 3> loc;
+       position(i, j, k, geomdata, loc);
+
+       Real h1 = 0.5e0_rt * geomdata.CellSize(0);
+       Real d1 = 1.e0_rt / (3.e0_rt * geomdata.CellSize(0));
+
+       r = loc[0];
+       r = d1 * (std::pow(r + h1, 3) - std::pow(r - h1, 3));
+
+#if AMREX_SPACEDIM >= 2
+       Real h2 = 0.5e0_rt * geomdata.CellSize(1);
+       Real d2 = 1.e0_rt / geomdata.CellSize(1);
+
+       s = loc[1];
+       s = d2 * (std::cos(s - h2) - std::cos(s + h2));
```

Comment:
Fortran code: https://github.com/AMReX-Astro/Castro/blob/development/Source/radiation/HABEC_nd.F90#L2494-L2552

We were doing
```
    if (dim >= 2) then
       d = 2
    else
       d = 1
    end if
```
and then this code was `s = loc(d);`. So my thinking was, first if we're in this ifdef block that means `AMREX_SPACEDIM >= 2`, so we were always evaluating to `d = 2`, and then decrementing by one for zero-indexing meant `d = 1`.

It's been a little while since I remember what these metrics actually mean so I can't say whether it makes sense, but I believe I've done the direct conversion from Fortran correctly.

Reply:
I found the original code here: https://github.com/AMReX-Astro/Castro/blob/a6f32d9221af323b5b71697824dce058f133aee7/Source/radiation/RAD_3D.F90#L42-L62 which also used `y` not `x` here.

Reply:
okay, I read it wrong.  In reality, this part will never be hit.